nengo_loihi/hardware/interface.py

```diff
@@ -56,31 +67,70 @@ def __init__(
     ):
         if isinstance(allocator, RoundRobin) and use_snips:
             raise SimulationError(
-                "snips are not supported for the " "RoundRobin allocator"
+                "snips are not supported for the RoundRobin allocator"
             )
 
         self.closed = False
-        self.use_snips = use_snips
-        self.check_nxsdk_version()
+        self.error_chip_map = {}  # maps synapses to chip locations for errors
 
-        self.nxsdk_board = None
-        self.nengo_io_h2c = None  # IO snip host-to-chip channel
-        self.nengo_io_c2h = None  # IO snip chip-to-host channel
         self._probe_filters = {}
         self._probe_filter_pos = {}
-        self._snip_probe_data = collections.OrderedDict()
-        self._chip2host_sent_steps = 0
 
-        # Maximum number of spikes that can be sent through
-        # the nengo_io_h2c channel on one timestep.
-        self.snip_max_spikes_per_step = snip_max_spikes_per_step
+        self.model = model
+        self.use_snips = use_snips
+        self.seed = seed
+
+        self.check_nxsdk_version()
+
+        validate_model(self.model)
 
         # clear cached content from SpikeProbe class attribute
         d_func(SpikeProbe, b"cHJvYmVEaWN0", b"Y2xlYXI=")
 
-        self.build(model, allocator=allocator, seed=seed)
+        # If we're using snips, tag all probes as being snip-based,
+        # as having normal probes at the same time as snips causes problems.
+        # This must be done before the build process to ensure information
+        # is stored properly.
+        if self.use_snips:
+            for block in self.model.blocks:
+                for probe in block.probes:
+                    probe.use_snip = True
```

Comment:
Does this mutate the underlying model? Would this cause an issue if the same model was run multiple consecutive times in separate simulators with separate snip settings?

Reply:
Oh nevermind this is a nengo-loihi probe from `block.py`. Not a core nengo probe.